### ISSUE TITLE
Immediate crash upon trying to run `louvre-views` inside Hyprland

### ISSUE THREAD

**Issue by ThatOneCalculator** [CuarzoSoftware/Louvre#31]
- Installed `louvre` from the AUR
- Ran `louvre-views`
```
amdgpu_device_initialize: amdgpu_query_info(ACCEL_WORKING) failed (-13)
amdgpu: amdgpu_device_initialize failed.
[1]    17391 segmentation fault (core dumped)  louvre-views
```

**Comment by ThatOneCalculator** [comment]
Coredump:

```
           PID: 28456 (louvre-views)
           UID: 1000 (kainoa)
           GID: 1001 (kainoa)
        Signal: 11 (SEGV)
     Timestamp: Sun 2024-02-11 15:54:47 PST (1min 4s ago)
  Command Line: louvre-views
    Executable: /usr/bin/louvre-views
 Control Group: /user.slice/user-1000.slice/session-1.scope
          Unit: session-1.scope
         Slice: user-1000.slice
       Session: 1
     Owner UID: 1000 (kainoa)
       Boot ID: 1e92a26a8c1249459f73919a5e96a26a
    Machine ID: 479b9a43f68e45d9b3e1d481d13c7920
      Hostname: anacapa
       Storage: /var/lib/systemd/coredump/core.louvre-views.1000.1e92a26a8c1249459f73919a5e96a26a.28456.1707695687000000.zst (present)
  Size on Disk: 2.3M
       Message: Process 28456 (louvre-views) of user 1000 dumped core.
                
                Stack trace of thread 28456:
                #0  0x00007d321a377fdd n/a (libc.so.6 + 0x159fdd)
                #1  0x00007d321a2bff02 __strdup (libc.so.6 + 0xa1f02)
                #2  0x00007d321791259d n/a (libEGL_mesa.so.0 + 0x2459d)
                #3  0x00007d321790392f n/a (libEGL_mesa.so.0 + 0x1592f)
                #4  0x00007d321adc9d96 _ZN6Louvre11LCompositor18LCompositorPrivate18initGraphicBackendEv (libLouvre.so.1 + 0x48d96)
                #5  0x00007d321adf0d20 _ZN6Louvre11LCompositor5startEv (libLouvre.so.1 + 0x6fd20)
                #6  0x0000584e46897225 n/a (louvre-views + 0xd225)
                #7  0x00007d321a243cd0 n/a (libc.so.6 + 0x25cd0)
                #8  0x00007d321a243d8a __libc_start_main (libc.so.6 + 0x25d8a)
                #9  0x0000584e46897585 n/a (louvre-views + 0xd585)
                
                Stack trace of thread 28459:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d321ae750ae n/a (libSRM.so.0 + 0x100ae)
                #3  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #4  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28460:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28462:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28461:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28463:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28464:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28467:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28465:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28466:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28468:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28469:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28470:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28472:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28471:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28473:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28482:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28481:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28479:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28491:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28492:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d3209f96b8c n/a (kms_swrast_dri.so + 0x196b8c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28489:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28480:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28477:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28475:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28474:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a79b684 n/a (kms_swrast_dri.so + 0x99b684)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28478:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28485:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28483:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28486:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28488:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28487:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28484:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28476:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                
                Stack trace of thread 28490:
                #0  0x00007d321a2a5ebe n/a (libc.so.6 + 0x87ebe)
                #1  0x00007d321a2a8750 pthread_cond_wait (libc.so.6 + 0x8a750)
                #2  0x00007d320a76f01c n/a (kms_swrast_dri.so + 0x96f01c)
                #3  0x00007d3209fe831c n/a (kms_swrast_dri.so + 0x1e831c)
                #4  0x00007d321a2a955a n/a (libc.so.6 + 0x8b55a)
                #5  0x00007d321a326a3c n/a (libc.so.6 + 0x108a3c)
                ELF object binary architecture: AMD x86-64
```

**Comment by 2xsaiko** [comment]
Oh hey, fancy seeing you here :^)

You need to run it from a TTY, it doesn't run as a nested compositor yet.

**Comment by ehopperdietzel** [comment]
Hi, currently, Louvre does not have a Wayland or X11 backend so you can't run it from Hyperland. But you can run it from a free virtual terminal (TTY). You can switch to a free TTY using the `chvt` command or by pressing `Ctrl + Alt + [F1, F2, etc]`. Also if you have another Wayland compositor running you should set the following envs to for example **WAYLAND_DISPLAY**=wayland-2 and **LOUVRE_WAYLAND_DISPLAY**=wayland-2 otherwise apps will be launched on Hyperland.

**Comment by ehopperdietzel** [comment]
@ThatOneCalculator, could you let me know if you were able to make it work? This way, I can proceed to close the issue.

**Comment by ThatOneCalculator** [comment]
It's very laggy+glitchy but works ^^

**Comment by ehopperdietzel** [comment]
Really? Which version of SRM are you using? You can see it by running it with **SRM_DEBUG**=4, there was a bug with version 0.5.1 but it should now be fixed with 0.5.2.

**Comment by ehopperdietzel** [comment]
I think the Arch SRM package is currently 0.5.1 so that must be it. You can build SRM manually following [this](https://cuarzosoftware.github.io/SRM/md_md__downloads.html)
